scipy/linalg/flapack_other.pyf.src

```diff
@@ -47,64 +47,144 @@ subroutine <prefix2>tgsen(ijob,wantq,wantz,select,n,a,lda,b,ldb,alphar,alphai,be
     callstatement (*f2py_func)(&ijob,&wantq,&wantz,select,&n,a,&lda,b,&ldb,alphar,alphai,beta,q,&ldq,z,&ldz,&m,&pl,&pr,dif,work,&lwork,iwork,&liwork,&info)
     callprotoargument F_INT*,F_INT*,F_INT*,F_INT*,F_INT*,<ctype2>*,F_INT*,<ctype2>*,F_INT*,<ctype2>*,<ctype2>*,<ctype2>*,<ctype2>*,F_INT*,<ctype2>*,F_INT*,F_INT*,<ctype2>*,<ctype2>*,<ctype2>*,<ctype2>*,F_INT*,F_INT*,F_INT*,F_INT*
 
-    integer intent(hide) :: ijob=4
-    integer intent(hide),check(wantq==0||wantq==1) :: wantq=1
-    integer intent(hide),check(wantz==0||wantz==1) :: wantz=1
-    logical intent(in),dimension(n) :: select
-    integer intent(hide),depend(a) :: n=shape(a,1)
-    <ftype2> intent(in,out,copy),dimension(lda,n) :: a
-    integer intent(hide),depend(a) :: lda=shape(a,0)
-    <ftype2> intent(in,out,copy),dimension(ldb,n) :: b
-    integer intent(hide),depend(b) :: ldb=shape(b,0)
-    <ftype2> intent(out),dimension(n) :: alphar
-    <ftype2> intent(out),dimension(n) :: alphai
-    <ftype2> intent(out),dimension(n) :: beta
-    <ftype2> intent(in,out,copy),dimension(ldq,n) :: q
-    integer intent(hide),depend(q) :: ldq=shape(q,0)
-    <ftype2> intent(in,out,copy),dimension(ldz,n) :: z
-    integer intent(hide),depend(z) :: ldz=shape(z,0)
+    integer optional, intent(in),check(ijob>=0&&ijob<=5) :: ijob=4
```

Comment:
The C code produced by this looks wrong. The else case is missing.

Reply:
That's handled by f2py. What should the else branch do?

Reply:
It should set and check ijob when ijob is set by the user. Right now this is missing and the next line, which is not supposed to be guarded by the else statement gets guarded by it, which is just wrong. It works correctly for all other methods so probably this code is wrong.

Reply:
There shouldn't be an else branch these evaluate to True or False and f2py keeps track of it. It is basically everywhere in the wrappers so I am not sure I get the point. Are you maybe following the compiler warnings ? Because you can get many warnings from the compilers about it.

Reply:
I was reading the code because of the other compiler warning, but for this one I was indeed wrong. ijob is set by `int_from_pyobj`. Sorry for the noise.

Reply:
No problem at all. Feedback always welcome